packages/ember-routing/lib/system/route.js

```diff
@@ -1979,6 +1985,10 @@ function buildRenderOptions(route, namePassed, name, options) {
     }
   }
 
+  if (options && indexOf(keys(options), 'outlet') !== -1 && typeof options.outlet === "undefined") {
+    throw new EmberError("You passed undefined as the outlet name.");
```

Comment:
@kiwiupover There is no outlet name in this case. What this is checking for is if someone passes undefined as the outlet name so the actual outlet name **is** called undefined